source/docs/software/advanced-controls/state-space/state-space-observers.rst

```diff
@@ -0,0 +1,118 @@
+    \text{Update step} \nonumber \\
+    \mathbf{K}_{k+1} &=
+        \mathbf{P}_{k+1}^- \mathbf{C}^T (\mathbf{C}\mathbf{P}_{k+1}^- \mathbf{C}^T +
+        \mathbf{R})^{-1} \\
+    \hat{\mathbf{x}}_{k+1}^+ &=
+        \hat{\mathbf{x}}_{k+1}^- + \mathbf{K}_{k+1}(\mathbf{y}_{k+1} -
+        \mathbf{C} \hat{\mathbf{x}}_{k+1}^- - \mathbf{D}\mathbf{u}_{k+1}) \\
+    \mathbf{P}_{k+1}^+ &= (\mathbf{I} - \mathbf{K}_{k+1}\mathbf{C})\mathbf{P}_{k+1}^-
+
+.. math::
+  \begin{array}{llll}
+    \mathbf{A} & \text{system matrix} & \hat{\mathbf{x}} & \text{state estimate vector} \\
+    \mathbf{B} & \text{input matrix}       & \mathbf{u} & \text{input vector} \\
+    \mathbf{C} & \text{output matrix}      & \mathbf{y} & \text{output vector} \\
+    \mathbf{D} & \text{feedthrough matrix} & \mathbf{\Gamma} & \text{process noise intensity vector} \\
+    \mathbf{P} & \text{error covariance matrix} & \mathbf{Q} & \text{process noise covariance matrix} \\
+    \mathbf{K} & \text{Kalman gain matrix} & \mathbf{R} & \text{measurement noise covariance matrix}
+  \end{array}
+
+The state estimate :math:`\mathbf{x}`, together with :math:`\mathbf{P}`, describe the mean and covariance of the Gaussian function that describes our filter's estimate of the system's true state.
+
+Process and Measurement Noise Covariance Matrices
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The process and measurement noise covariance matrices :math:`\mathbf{Q}` and :math:`\mathbf{R}` describe the variance of each of our states and measurements. Remember that for a Gaussian function, variance is the square of the function's standard deviation. In a WPILib, Q and R are diagonal matrices whose diagonals contain their respective variances. For example, a Kalman filter with states :math:`\begin{bmatrix}\text{position} \\ \text{velocity} \end{bmatrix}` and measurements :math:`\begin{bmatrix}\text{position} \end{bmatrix}` with state standard deviations :math:`\begin{bmatrix}0.1 \\ 1.0\end{bmatrix}` and measurement standard deviation :math:`\begin{bmatrix}0.01\end{bmatrix}` would have the following :math:`\mathbf{Q}` and :math:`\mathbf{R}` matrices:
+
+.. math::
+  Q = \begin{bmatrix}0.01 0 \\ 0 1.0\end{bmatrix},
+  R = \begin{bmatrix}0.0001\end{bmatrix}
+
+Error Covariance Matrix
+^^^^^^^^^^^^^^^^^^^^^^^
+
+The error covariance matrix :math:`\mathbf{P}` describes the covariance of the state estimate :math:`\mathbf{\hat{x}}`. Informally, :math:`\mathbf{P}` describes our certainty about the estimated :term:`state`. If :math:`\mathbf{P}` is large, our uncertainty about the true state is large. Conversely, a :math:`\mathbf{P}` with smaller elements would imply less uncertainty about our true state. In the prediction step, :math:`\mathbf{P}` grows at a rate proportional to the process noise covariance :math:`\mathbf{Q}` and process noise intensity vector :math:`\mathbf{\Gamma}` to show how our certainty about the system's state decreases as we project the model forward.
```

Comment:
During prediction, the error covariance increases quadratically because it's `APA^T`. Random walks look like `dx/dt = Ax + Bu + w`, which increases the standard deviation linearly with time. Since the variance is the square of the standard deviation, the variance increases quadratically.

Reply:
![Screenshot_2020-09-20_20-50-10](https://user-images.githubusercontent.com/2748555/93731743-32376480-fb83-11ea-8d79-43ed0e7b54cc.png)